controllers/monitoring_alert.go

```diff
@@ -303,14 +319,136 @@ func (r *ManagedMCGReconciler) reconcileAlertmanager() error {
 	return nil
 }
 
+// AlertRelabelConfigSecret will have configuration for relabeling the alerts that are firing.
+// It will add namespace label to firing alerts before they are sent to the alertmanager.
+func (r *ManagedMCGReconciler) reconcileAlertRelabelConfigSecret() error {
+	r.Log.Info("Reconciling alertRelabelConfigSecret")
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.alertRelabelConfigSecret, func() error {
+		if err := r.own(r.alertRelabelConfigSecret); err != nil {
+			return err
+		}
+		alertRelabelConfig := []struct {
+			TargetLabel string `yaml:"target_label,omitempty"`
+			Replacement string `yaml:"replacement,omitempty"`
+		}{{
+			TargetLabel: "namespace",
+			Replacement: r.namespace,
+		}}
+		config, err := yaml.Marshal(alertRelabelConfig)
+		if err != nil {
+			return fmt.Errorf("unable to encode alert relabel conifg: %w", err)
+		}
+		r.alertRelabelConfigSecret.Data = map[string][]byte{
+			alertRelabelConfigSecretKey: config,
+		}
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create/update AlertRelabelConfigSecret: %w", err)
+	}
+
+	return nil
+}
+
+func (r *ManagedMCGReconciler) reconcileKubeRBACConfigMap() error {
+	r.Log.Info("Reconciling kubeRBACConfigMap")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.kubeRBACConfigMap, func() error {
+		if err := r.own(r.kubeRBACConfigMap); err != nil {
+			return err
+		}
+
+		r.kubeRBACConfigMap.Data = templates.KubeRBACProxyConfigMap.DeepCopy().Data
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create kubeRBACConfig config map: %w", err)
+	}
+
+	return nil
+}
+
+// reconcilePrometheusService function wait for prometheus Service
+// to start and sets appropriate annotation for 'service-ca' controller.
+func (r *ManagedMCGReconciler) reconcilePrometheusService() error {
+	r.Log.Info("Reconciling PrometheusService")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.prometheusService, func() error {
+		if err := r.own(r.prometheusService); err != nil {
+			return err
+		}
+
+		r.prometheusService.Spec.Ports = []corev1.ServicePort{
+			{
+				Name:       "https",
+				Protocol:   corev1.ProtocolTCP,
+				Port:       int32(templates.KubeRBACProxyPortNumber),
+				TargetPort: intstr.FromString("https"),
+			},
+		}
+		r.prometheusService.Spec.Selector = map[string]string{
+			"app.kubernetes.io/name": r.prometheusService.Name,
+		}
+		utils.AddAnnotation(
```

Comment:
we are already creating cert for the console using "service.alpha.openshift.io/serving-cert-secret-name", can we reuse that cert in here, instead of creating a new one for the Prometheus
https://github.com/red-hat-storage/mcg-osd-deployer/blob/main/console/console.go#L44

Reply:
Is there a down side to using two different secrets for different purposes? @v-harihar and I talked about this a while ago and we agreed that doing this might give rise to race conditions, and since having teo different secrets for Prometheus and the UI plugin didn't affect us adversely, we decided to proceed with that.

Reply:
@rexagod  how it will give a race condition? its a kind of secret only right
If we can use one cert for both console and Prometheus, do we need to create two different resources for it

Reply:
If the resource is requested by one part of the loop before it was created by the other part of it, then a race condition will arise. Reconciliation loops need to ensure that they work as expected (are able to reconcile themselves) when triggered at any point of time (with whatever configuration they may have then).

Creating 2 secrets will in no way increase the complexity since it's a linear increase, not exponential one. Again, we will introduce this flaky (race condition) behavior by using one secret, but we can avoid it entirely by using two, also given that these secrets are in no way similar in purpose or nature to each other.

There is a downside to using one secret, but none when using two.

Reply:
@rexagod cert creation is an expensive process I wanted to avoid this if possible, for now, I think we can go ahead with this and we need to test reconcile with one cert and test what the impact

Reply:
cc @v-harihar